ehcache-impl/src/main/java/org/ehcache/impl/internal/loaderwriter/writebehind/BatchingLocalHeapWriteBehindQueue.java

```diff
@@ -188,14 +191,23 @@ public boolean add(SingleOperation<K, V> operation) {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public void run() {
       try {
         List<BatchOperation<K, V>> batches = createMonomorphicBatches(operations());
         // execute the batch operations
         for (BatchOperation<K, V> batch : batches) {
           try {
             batch.performOperation(cacheLoaderWriter);
+          } catch (BulkCacheWritingException e) {
+            for (Object key : e.getFailures().keySet()) {
+              keyCleanUpMethod.accept((K) key);
```

Comment:
Doesn't the same problem with the ordering of the cleanup call and the `latest.remove(...)` call happen in the batching queue too?

Reply:
Hi Chris,

I have incorporated the review comments.